test-suite/xoshiro256starstar.cpp

```diff
@@ -0,0 +1,152 @@
+/* -*- mode: c++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*
+ Copyright (C) 2023 Ralf Konrad Eckel
+
+ This file is part of QuantLib, a free-software/open-source library
+ for financial quantitative analysts and developers - http://quantlib.org/
+
+ QuantLib is free software: you can redistribute it and/or modify it
+ under the terms of the QuantLib license.  You should have received a
+ copy of the license along with this program; if not, please email
+ <quantlib-dev@lists.sf.net>. The license is also available online at
+ <http://quantlib.org/license.shtml>.
+
+ This program is distributed in the hope that it will be useful, but WITHOUT
+ ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ FOR A PARTICULAR PURPOSE.  See the license for more details.
+*/
+
+#include "xoshiro256starstar.hpp"
+#include "utilities.hpp"
+#include <ql/math/randomnumbers/xoshiro256starstaruniformrng.hpp>
+#include <numeric>
+
+// we do not want to change the original xoshiro256starstar.c implementation. Therefore, we suppress
+// any warnings from this file.
+// clang-format off
+#if defined(__GNUC__)
+    _Pragma("GCC diagnostic push")
+    _Pragma("GCC diagnostic ignored \"-Wsign-compare\"")
+#elif defined(__clang__)
+    _Pragma("clang diagnostic push")
+    _Pragma("clang diagnostic ignored \"-Wsign-compare\"")
+#endif
+
+extern "C" {
+#include "xoshiro256starstar.c"
```

Comment:
Yes, as you say, it looks a bit weird.  We might just check a few samples instead of 1000 (e.g. the 1st, the 10th, the 100th, the 1000th...) and those we could just hard-code in the test.

Also, if we actually used the file, it should be added somehow to the Makefile so that it's included in the released tarball.  But I'd sidestep this.

Reply:
I also thought about _check a few samples_ like in the Mersenne Twister test. However, in there is a reference where the numbers are coming from.

Or fully inline `xoshiro256starstar.c`?
```c++
#endif

extern "C" {
/*  Written in 2018 by David Blackman and Sebastiano Vigna (vigna@acm.org)

To the extent possible under law, the author has dedicated all copyright
and related and neighboring rights to this software to the public domain
worldwide. This software is distributed without any warranty.

See <http://creativecommons.org/publicdomain/zero/1.0/>. */

#include <stdint.h>

/* This is xoshiro256** 1.0, one of our all-purpose, rock-solid
   generators. It has excellent (sub-ns) speed, a state (256 bits) that is
   large enough for any parallel application, and it passes all tests we
   are aware of.

   For generating just floating-point numbers, xoshiro256+ is even faster.

   The state must be seeded so that it is not everywhere zero. If you have
   a 64-bit seed, we suggest to seed a splitmix64 generator and use its
   output to fill s. */

static inline uint64_t rotl(const uint64_t x, int k) {
	return (x << k) | (x >> (64 - k));
}


static uint64_t s[4];

uint64_t next(void) {
	const uint64_t result = rotl(s[1] * 5, 7) * 9;

	const uint64_t t = s[1] << 17;

	s[2]...
...
}
```

Reply:
Sure, why not.